policy_set_version.go

```diff
@@ -0,0 +1,158 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ PolicySetVersions = (*policySetVersions)(nil)
+
+// PolicySetVersions describes all the Policy Set Version related methods that the Terraform
+// Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/policy-sets.html#create-a-policy-set-version
+type PolicySetVersions interface {
+	// Create is used to create a new Policy Set Version.
+	Create(ctx context.Context, policySetID string) (*PolicySetVersion, error)
+
+	// Read is used to read a Policy Set Version by its ID.
+	Read(ctx context.Context, policySetVersionID string) (*PolicySetVersion, error)
+
+	// Upload uploads policy files. It takes a Policy Set Version and a path
+	// to the set of sentinel files, which will be packaged by hashicorp/go-slug
+	// before being uploaded.
+	Upload(ctx context.Context, psv PolicySetVersion, path string) error
+}
+
+// policySetVersions implements PolicySetVersions.
+type policySetVersions struct {
+	client *Client
+}
+
+// PolciySetVersionSource represents a source type of a policy set version.
+type PolciySetVersionSource string
```

Comment:
For this and several other places in these changes, `s/ Polciy/Policy/` 😄  I'm impressed by the consistency, though!

Reply:
Lots of vim copy/pasting 😅 . Will update